### ISSUE TITLE
Add missing <main> elements to template patterns

### ISSUE THREAD

**Issue by carolinan** [WordPress/twentytwentyfive#278]
**Description**

These templates are missing the `<main>` element and it needs to be added:

- [x] news-blog-archive-template.php, News blog archive
- [x] news-blog-home-template.php,  News blog home
- [x] news-blog-single-post-with-sidebar-template.php News blog single post with sidebar
- [x] news-blog-with-featured-posts-grid-template.php, News blog with featured posts grid
- [x] news-blog-with-sidebar-template.php News blog with sidebar
- [x] news-blog-search-results-template.php
- [x] offset-post-no-featured-image-template.php Offset post without featured image
- [x] vertical-header-right-aligned-archive-template, Archive for the right aligned blog
- [x] vertical-header-right-aligned-home-template.php Homepage for right aligned blog
- [x] vertical-header-right-aligned-page-template.php  Page template for the right aligned blog
- [x] post-with-left-aligned-content.php
- [x] vertical-header-right-aligned-posts.php
- [x] vertical-header-right-aligned-search-results-template.php

**Expected behavior**

All pages (templates) has one `<main>` element.

All templates must include main, both because all content needs to be inside a landmark, but also because in block themes, the automated skip link requires it.
https://make.wordpress.org/themes/handbook/review/accessibility/required/#aria-landmark-roles

**Comment by carolinan** [comment]
When working on these, please submit smaller pull requests since they will be easier to review and test.

**Comment by noruzzamans** [comment]
@carolinan I am interested in working on these issues. Could you please share an example of how to resolve them? Thanks

**Comment by KevinTaron** [comment]
I've added a pull request for all templates with named veritcal-header-right...
#280 

@noruzzamans maybe you can look into my pull request — that's an example how to do it. 
Probably you should wait and see if the maintainers accept the pull request as it is. ;)

**Comment by carolinan** [comment]
> @carolinan I am interested in working on these issues. Could you please share an example of how to resolve them? Thanks

Some patterns needs a new block inserted, that will be used as the main, others need the HTML tag on existing groups to be updated.

When changing a group block's HTML tag from `<div>`, to `<main>`, both the block comment (the JSON-style attributes) and the  tag on the block itself needs to be updated.

From
```
<!-- wp:group {"layout":{"type":"constrained"}} -->
<div class="wp-block-group"></div>
<!-- /wp:group -->
```
To

```
<!-- wp:group {"tagName":"main","layout":{"type":"constrained"}} -->
<main class="wp-block-group"></main>
<!-- /wp:group -->
```

**Comment by carolinan** [comment]
Time permitting, we could reconsider if we really want the `<main>` to wrap everything between the header and footer.
It can be a better practice to wrap the `<main>` around the **actual main content** and use other landmarks for post navigation and comments.

**Comment by KevinTaron** [comment]
> Time permitting, we could reconsider if we really want the `<main>` to wrap everything between the header and footer.
> It can be a better practice to wrap the `<main>` around the **actual main content** and use other landmarks for post navigation and comments.

Actually I think that's a good idea.

**Comment by carolinan** [comment]
I am waiting with the news templates until the comment sections are added to avoid more merge conflicts.

**Comment by carolinan** [comment]
..and there the comments are merged 😄

**Comment by juanfra** [comment]
Looks like this one is complete. Can you confirm, @carolinan?

**Comment by carolinan** [comment]
I want to do another round of testing, some of the headings regressed, so I want to make sure.

**Comment by juanfra** [comment]
Perfect, thank you!

**Comment by carolinan** [comment]
All looks good to me.
